### ISSUE TITLE
Make sure net/api/client.Subscribe loop ends gracefully when client is closing

### ISSUE THREAD

**Issue by merlinran** [textileio/go-threads#515]
When `client.Close` is called, the gRPC connection gets closed, and the `stream.Recv()` call here returns a gRPC internal error [`ErrConnClosing`](https://github.com/grpc/grpc-go/blob/9572fd6faeaee33fe295ce3a79eab729d05bb349/internal/transport/transport.go#L750), which is not considered a grace close, hence causes a `log.Fatalf()`.

> 2021/05/14 13:07:05 error in subscription stream: rpc error: code = Unavailable desc = transport is closing

https://github.com/textileio/go-threads/blob/5d09258e40a4e008cab2d4de9bbeb6ffd5357054/net/api/client/client.go#L297-L307

This shouldn't be considered a fatal error, but maybe more importantly, provide a way to signal the loop to return immediately?

**Comment by sanderpick** [comment]
could we check `errors.Is(err, ErrConnClosing)` and just bail when `true`?

**Comment by merlinran** [comment]
Nope, `ErrConnClosing` is defined in an internal package so we don't have access. Didn't check if the error wrapper can be recognizable though. Anyway it should be low priority since it only happens when closing down.

**Comment by sanderpick** [comment]
Cool, I thought that since it was capitalized it was exported. Anyway, sgtm!
